src/class/CgiManager/CgiManager.cpp

```diff
@@ -0,0 +1,176 @@
+					   const std::string &port)
+	: _basicHTTPRequest(basicHTTPRequest)
+	, _pathCGI(pathCGI)
+	, _serverName(serverName)
+	, _port(port)
+{
+	_env = Tab();
+	_argv = Tab();
+	_pipe = Pipe();
+	_basicHTTPRequest.parseRaw();
+	if (_basicHTTPRequest.isBody())
+		_basicHTTPRequest.parseBody();
+}
+
+const char *CgiManager::CgiManagerException::what() const throw()
+{
+	return "Error in CgiManager";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+CgiManager::~CgiManager()
+{
+}
+
+/*
+** --------------------------------- OVERLOAD ---------------------------------
+*/
+
+std::ostream &operator<<(std::ostream &o, CgiManager const &i)
+{
+	o << "CgiManager[ ServerName=\"" << i.getServerName() << "\""
+	  << "Port=\"" << i.getPort() << "\""
+	  << "PathCGI=\"" << i.getPathCGI() << "\""
+	  << "BasicHTTPRequest=\"" << i.getBasicHTTPRequest() << "\""
+	  << "]";
+	return o;
+}
+
+/*
+** --------------------------------- METHODS ----------------------------------
+*/
+
+void CgiManager::_setEnv(void)
+{
+	std::map<std::string, std::string> env;
+	if (_basicHTTPRequest.getHeaders().find(httpConstants::CONTENT_LENGTH) != _basicHTTPRequest.getHeaders().end())
+		env[httpConstants::CONTENT_LENGTH] = _basicHTTPRequest.getHeaders().at(httpConstants::CONTENT_LENGTH);
+	if (_basicHTTPRequest.getHeaders().find(httpConstants::CONTENT_TYPE_FIELD) != _basicHTTPRequest.getHeaders().end())
+		env[httpConstants::CONTENT_TYPE_FIELD] = _basicHTTPRequest.getHeaders().at(httpConstants::CONTENT_TYPE_FIELD);
+	env[httpConstants::REQUEST_METHOD] = _basicHTTPRequest.getType();
+	env[httpConstants::SCRIPT_FILENAME] = _pathCGI.get();
+	if (_basicHTTPRequest.getQuery().length() > 1)
+		env[httpConstants::QUERY_STRING] = _basicHTTPRequest.getQuery().substr(1);
+	env[httpConstants::SERVER_PROTOCOL] = _basicHTTPRequest.getHTTPVersion();
+	env[httpConstants::SERVER_PORT] = _port;
+	env[httpConstants::SERVER_NAME] = _serverName;
+	env[httpConstants::PATH_INFO] = _basicHTTPRequest.getPath();
+
+	std::map<std::string, std::string>::const_iterator it = env.begin();
+	for (int i = 0; it != env.end(); it++, i++)
+		_env.add(it->first + "=" + it->second);
+}
+
+void CgiManager::_setArgv(const Path &pathServer)
+{
+	_argv.add(_pathCGI.get());
+	_argv.add(pathServer.get() + _basicHTTPRequest.getPath());
+}
+
+int CgiManager::_createFork(void)
+{
+	int pid = fork();
+	if (pid < ERROR)
+		throw CgiManager::CgiManagerException();
+	return (pid);
+}
+
+void CgiManager::_childProcess(void)
+{
+	int _exit_status;
+	_pipe.setChild();
+	_exit_status = execve(_argv.toCTable()[0], _argv.toCTable(), _env.toCTable());
+	if (_exit_status < 0)
+		throw CgiManager::CgiManagerException();
+	exit(_exit_status);
+}
+
+std::string CgiManager::_parentProcess(PollManager poll)
+{
+	_pipe.setParent();
+	if (_basicHTTPRequest.isBody() && poll.isAvailable(_pipe.getInput(), POLLOUT))
```

Comment:
i think the way you use poll() here will cause us trouble in the future.
but until we have a clear structure of the server flow i think we can leave it like this.